vitest.setup.ts

```diff
@@ -8,6 +8,368 @@ import parseLinksHeader from 'parse-link-header';
+    getOutput(): Promise<any>;
+    getConnectionData(): Promise<any>;
+    getMetadataData(): Promise<any>;
+    getCachedResponse(identity: ConfigIdentity): Promise<any>;
+    getUpdateMetadata(): Promise<any>;
+    getDeleteRecordsFromPreviousExecutions(): Promise<any>;
+}
+
+class LegacyMockLoader implements MockLoaderStrategy {
+    constructor(
+        private dirname: string,
+        private name: string
+    ) {}
+
+    private async getMockFile(fileName: string, throwOnMissing: boolean, identity?: ConfigIdentity) {
+        const filePath = path.resolve(this.dirname, `../mocks/${fileName}.json`);
+        try {
+            const fileContent = await fs.readFile(filePath, 'utf-8');
+            const data = JSON.parse(fileContent);
+            return data;
+        } catch (error) {
+            if (throwOnMissing) {
+                throw new Error(`Failed to load mock data from ${filePath}: ${error.message} ${identity ? JSON.stringify(identity, null, 2) : ''}`);
+            }
+        }
+    }
+
+    private async hashDirExists(hashDir: string) {
+        const filePath = path.resolve(this.dirname, `../mocks/${hashDir}/`);
+
+        try {
+            await fs.stat(filePath);
+            return true;
+        } catch (error) {
+            return false;
+        }
+    }
+
+    async getCachedResponse(identity: ConfigIdentity) {
+        const dir = `nango/${identity.method}/proxy/${identity.endpoint}/${this.name}/`;
+        const hashBasedPath = `${dir}/${identity.requestIdentityHash}`;
+
+        if (await this.hashDirExists(dir)) {
+            const data = await this.getMockFile(hashBasedPath, true, identity);
+            return data;
+        } else {
+            return { response: await this.getMockFile(`nango/${identity.method}/proxy/${identity.endpoint}/${this.name}`, true, identity) };
+        }
+    }
+
+    async getBatchSaveData(modelName: string) {
+        return this.getMockFile(`${this.name}/${modelName}/batchSave`, true);
+    }
+
+    async getBatchDeleteData(modelName: string) {
+        return this.getMockFile(`${this.name}/${modelName}/batchDelete`, true);
+    }
+
+    async getInput() {
+        return this.getMockFile(`${this.name}/input`, false);
+    }
+
+    async getOutput() {
+        return this.getMockFile(`${this.name}/output`, true);
+    }
+
+    async getConnectionData() {
+        return this.getMockFile(`nango/getConnection`, true);
+    }
+
+    async getMetadataData() {
+        return this.getMockFile('nango/getMetadata', true);
+    }
+
+    async getUpdateMetadata() {
+        return this.getMockFile('nango/updateMetadata', false);
+    }
+
+    async getDeleteRecordsFromPreviousExecutions() {
+        return this.getMockFile('nango/deleteRecordsFromPreviousExecutions', false);
+    }
+}
+
+class UnifiedMockLoader implements MockLoaderStrategy {
+    private mockData: any;
+
+    constructor(private mockFilePath: string) {}
+
+    private async loadMockFile() {
```

Comment:
does it bring anything to load the data dynamically instead of when instantiating the loader?